sdk/messaging/azservicebus/sender.go

```diff
@@ -57,18 +57,14 @@ func MessageBatchWithMaxSize(maxSizeInBytes int) func(options *messageBatchOptio
 // messages. Sending a batch of messages is more efficient than sending the
 // messages one at a time.
 func (s *Sender) NewMessageBatch(ctx context.Context, options ...MessageBatchOption) (*MessageBatch, error) {
-	if s.linkState.Closed() {
-		return nil, s.linkState.Err()
-	}
-
-	legacySender, err := s.createAmqpSender(ctx)
+	sender, _, _, _, err := s.links.Get(ctx)
```

Comment:
The dogsled makes me think this method needs some refactoring (note, I haven't looked at it yet).

Reply:
The easiest one would be to just return a struct, I think.

Reply:
I can change this in the next PR.